### DISCUSSION TITLE
Change pointcloud opacity with pointdata dict

### DISCUSSION THREAD

**Discussion by jkissi** [marcomusy/vedo#737] [Q&A] [answered]
Hi Marco, hope you are well!! 
As the title suggests, I would like to change the opacity of a selection of points in an already created pointcloud. Currently I have code that looks like this:
```
colour_array = self.fuzzy['vars']['colour_array']
# currently the array is three dimensional (one column for each colour channel)
carraygt0 =  colour_array.copy()
alpha = np.full((len(carraygt0), 1), 255)
# append an alpha channel (I assume it has to be an int as well???)
carraygt0 = np.append(carraygt0, alpha, axis=1)
maskgt0 = (cloud_df['in_gt'] == 0)
# mask the colour and the opacity. (it this the right way to do it???)
carraygt0[maskgt0] = [255, 215, 0, 200]
self.pcds[0].pointdata['gt0'] = carraygt0

```
What happens here is that the colour is change only, rather than changing the colour and opacity when the pointdata field is selected. 
I there a way to change the colour and opacity of a selection of points using the pointdata dict?

Cheers 
j

**Comment by marcomusy** [comment, marked as answer]
Hi J
points are also cells by themselves so, check out
https://github.com/marcomusy/vedo/blob/master/examples/basic/colorMeshCells.py

```python
from vedo import *

#settings.use_depth_peeling = True

pts = Sphere().clean().points()
tor = Points(pts, r=20)
#tor.render_points_as_spheres(False)

rgba = np.random.rand(tor.ncells, 4)*255 # RGBA values

colorlist = rgba.astype(np.uint8)  # note the uint8 type
tor.celldata["mycols"] = colorlist
tor.celldata.select("mycols")

show(tor).close()
```
<img width="249" alt="Screenshot 2022-11-20 at 13 54 55" src="https://user-images.githubusercontent.com/32848391/202902989-e811c064-fb2d-473f-8678-b2039eb184b4.png">

Also note the shortcut:
```python
# ...
rgba = np.random.rand(tor.ncells, 4)*255 # RGBA values
tor.cell_individual_colors(rgba)

show(tor).close()
```

**Reply by jkissi** [reply]
It worked perfectly!! thanks marco!! 👍
